src/types/inet.rs

```diff
@@ -0,0 +1,70 @@
+use napi::{bindgen_prelude::Buffer, Error, Status};
+use scylla::frame::response::result::CqlValue;
+use std::net::IpAddr;
+#[napi]
+pub struct InetAddressWrapper {
+    inet: IpAddr,
+}
+
+#[napi]
+impl InetAddressWrapper {
+    #[napi]
+    pub fn new(buffer: Buffer) -> napi::Result<Self> {
+        let buffer: Vec<u8> = buffer.into();
+        if buffer.len() == 4 {
+            let mut ip = [0u8; 4];
+            ip.copy_from_slice(&buffer);
+            Ok(Self {
+                inet: IpAddr::from(ip),
+            })
+        } else if buffer.len() == 16 {
+            let mut ip = [0u8; 16];
+            ip.copy_from_slice(&buffer);
+            Ok(Self {
+                inet: IpAddr::from(ip),
+            })
+        } else {
+            return Err(Error::new(
+                Status::GenericFailure,
+                "Invalid IP address length",
+            ));
+        }
```

Comment:
Right, I feared this, too :smile:
So another suggestion:
instead of `if 4 == len then copy_from_slice else if 16 == len then ...`
let's use `TryFrom<[u8]> for [u8; n]` impl:
```rust
        if let Ok(arr) = [u8; 4]::try_from(buffer) {
            Ok(Self {
                inet: IpAddr::from(ip),
            })
        } else if let Ok(arr) = [u8; 16]::try_from(buffer) {
            Ok(Self {
                inet: IpAddr::from(ip),
            })
        } else {
            Err(Error::new(
                Status::GenericFailure,
                "Invalid IP address length",
            ));
        }